src/screens/AccountNetworkChooser.js

```diff
@@ -16,118 +16,241 @@
+	const renderShowMoreButton = () => {
+		if (isNew) return;
+		if (!shouldShowMoreNetworks) {
+			return (
+				<>
+					<AccountCard
+						isAdd={true}
+						isNetworkCard={true}
+						onPress={() => setShouldShowMoreNetworks(true)}
+						title="Add Network Account"
+						networkColor={colors.bg}
+					/>
+					<Separator style={{ backgroundColor: 'transparent', height: 120 }} />
+				</>
+			);
+		}
+	};
 
-		if (!__DEV__) {
-			excludedNetworks.push(SubstrateNetworkKeys.SUBSTRATE_DEV);
-			excludedNetworks.push(SubstrateNetworkKeys.KUSAMA_DEV);
+	const renderScreenHeading = () => {
+		if (isNew) {
+			return <ScreenHeading title={'Create your first Keypair'} />;
+		} else if (shouldShowMoreNetworks) {
+			return (
+				<ScreenHeading
+					title={'Choose Network'}
+					onPress={() => setShouldShowMoreNetworks(false)}
+				/>
+			);
 		}
+	};
 
+	const renderScanButton = () => {
+		if (isNew) return;
+		else if (shouldShowMoreNetworks) return;
 		return (
-			<ScrollView style={styles.body} contentContainerStyle={{ padding: 20 }}>
-				<Text style={styles.title}>CHOOSE NETWORK</Text>
-				{Object.entries(NETWORK_LIST)
-					.filter(([networkKey]) => !excludedNetworks.includes(networkKey))
-					.map(([networkKey, networkParams]) => (
-						<TouchableItem
-							key={networkKey}
-							style={[
-								styles.card,
-								{
-									backgroundColor: networkParams.color,
-									marginTop: 20
-								}
-							]}
-							onPress={() => {
-								accounts.updateNew(empty('', networkKey));
-								navigation.goBack();
-							}}
-						>
-							<Text
-								style={[
-									styles.cardText,
-									{
-										color: networkParams.secondaryColor
-									}
-								]}
-							>
-								{networkParams.title}
-							</Text>
-						</TouchableItem>
-					))}
-			</ScrollView>
+			<ButtonMainAction
+				testID={testIDs.AccountNetworkChooser.scanButton}
+				title={'Scan'}
+				onPress={() => navigation.navigate('QrScanner')}
+			/>
 		);
-	}
+	};
+
+	const onNetworkChosen = async (networkKey, networkParams) => {
+		if (isNew) {
+			if (networkParams.protocol === NetworkProtocols.SUBSTRATE) {
+				await deriveSubstrateDefault(networkKey, networkParams);
+			} else {
+				await deriveEthereumAccount(networkKey);
+			}
+		} else {
+			const paths = Array.from(currentIdentity.meta.keys());
+			const listedPaths = getPathsWithSubstrateNetwork(paths, networkKey);
+			if (networkParams.protocol === NetworkProtocols.SUBSTRATE) {
+				if (listedPaths.length === 0)
+					return navigation.navigate('PathDerivation', {
+						networkKey
+					});
+			} else if (!paths.includes(networkKey)) {
+				return await deriveEthereumAccount(networkKey);
+			}
+			navigation.navigate('PathsList', { networkKey });
+		}
+	};
+
+	if (!loaded) return <ScrollView style={styles.body} />;
+
+	if (identities.length === 0) return showOnboardingMessage();
+
+	const NetworkList = Object.entries(NETWORK_LIST).filter(getNetworkKeys);
```

Comment:
was there a special reason why this is not camelCase?

Reply:
typo here